nucleus/payara-modules/hazelcast-bootstrap/src/main/java/fish/payara/nucleus/cluster/MemberEvent.java

```diff
@@ -39,31 +39,33 @@
  */
 package fish.payara.nucleus.cluster;
 
-import com.hazelcast.core.Member;
+import com.hazelcast.cluster.Member;
 import fish.payara.nucleus.hazelcast.HazelcastCore;
+import java.util.UUID;
 
 /**
  *
  * @author Steve Millidge (Payara Foundation)
  */
 public class MemberEvent {
     
-    public MemberEvent(Member member) {
+    public MemberEvent(HazelcastCore hzCore, Member member) {
+        this.hzCore = hzCore;
         this.member = member;
     }
     
-    public String getUuid() {
+    public UUID getUuid() {
         return member.getUuid();
     }
     
     public String getServer() {
-        return member.getStringAttribute(HazelcastCore.INSTANCE_ATTRIBUTE);
+        return hzCore.getAttribute(member.getUuid(), HazelcastCore.INSTANCE_ATTRIBUTE);
     }
     
     public String getServerGroup() {
-        return member.getStringAttribute(HazelcastCore.INSTANCE_GROUP_ATTRIBUTE);
+        return member.getAttribute(HazelcastCore.INSTANCE_GROUP_ATTRIBUTE);
     }
     
-    private Member member;
-    
+    private final HazelcastCore hzCore;
```

Comment:
Could you move those fields to the beginning of the class?

Reply:
Out of scope, again

Reply:
Improving code quality by trivial steps is in the scope. But it depends on your decision, I just noticed it ;)